src/YesSql.Core/Session.cs

```diff
@@ -765,6 +765,12 @@ private async Task FlushInternalAsync(bool saving)
             }
             catch
             {
+                // Only check thread-safety if not called from SaveChangesAsync
+                if (!saving)
+                {
+                    ExitAsyncExecution();
```

Comment:
You can't just do that, because now `ExitAsyncExecution` is called twice. I would propose a `CancelInternalAsync` which avoids calling `Enter-/ExitAsyncExecution` at all.

Reply:
The mustExitAsyncExecution will prevent the ExitAsyncExecution  from being called more that needed

Reply:
Sorry just saw both of you comments. So you good with this approach?

Reply:
Yep, that wasn't there at first 🤭🙈

Reply:
> Sorry just saw both of you comments. So you good with this approach?

Yes, LGTM